File: src/main/java/dev/chargedbyte/wim/model/Product.java
package dev.chargedbyte.wim.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.keyvalue.annotation.KeySpace;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@KeySpace("product")
public class Product {
    @Id
    private String id;
    private Category category;
    private String name;
    private List<String> colors;
    private Integer price;
    private String manufacturer;
    private Availability availability;
}

File: app/src/model/product.ts
export enum Category {
  Unknown,
  Beanies,
  Facemasks,
  Gloves,
}

export enum Availability {
  Unknown,
  InStock,
  LessThan10,
  OutOfStock,
}

export default interface Product {
  id: string
  category: Category
  name: string
  colors: Array<string>
  price: number
  manufacturer: string
  availability: Availability
}
